core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -167,59 +154,52 @@ public TableScan asOfTime(long timestampMillis) {
   @Override
   public TableScan option(String property, String value) {
     ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
-    builder.putAll(options);
+    builder.putAll(context.options());
     builder.put(property, value);
 
     return newRefinedScan(
-        ops, table, snapshotId, schema, rowFilter, ignoreResiduals,
-        caseSensitive, colStats, selectedColumns, builder.build());
+        ops, table, schema, context.options(builder.build()));
   }
 
   @Override
   public TableScan project(Schema projectedSchema) {
     return newRefinedScan(
-        ops, table, snapshotId, projectedSchema, rowFilter, ignoreResiduals,
-        caseSensitive, colStats, selectedColumns, options);
+        ops, table, projectedSchema, context.copy());
```

Comment:
What is your rationale for not making `projectedSchema` part of `context`?

Reply:
Schema does not have a `default` value like  the elements in the context, hence it would always need to be specified when creating the context.
In this refinement call, `projectedSchema` is not an additional context field but takes over schema in `TableScan`.